client/layout/guided-tours/config-elements/quit.js

```diff
@@ -28,6 +30,42 @@ export default class Quit extends Component {
 		super( props, context );
 	}
 
+	componentDidMount() {
+		this.addTargetListener();
+	}
+
+	componentWillUnmount() {
+		this.removeTargetListener();
+	}
+
+	componentWillUpdate() {
```

Comment:
Should we check if our target has changed before toggling listeners?

Reply:
To be fair, I could also even remove both `didUpdate` and `willUpdate` methods (also, the latter is deprecated!), and this should keep working.
Although, I'd avoid making any assumptions: those methods are there in the [`Continue`](https://github.com/Automattic/wp-calypso/blob/master/client/layout/guided-tours/config-elements/continue.js#L48-L54) component as well, maybe they help with some corner cases? (cc @lsinger)

Reply:
I vaguely remember that a bunch of problems stemmed from assuming that only mounting and unmounting would matter, when in reality there'd be a bunch of updates that would pull the carpet from under our / the component's feet. I'd favor keeping these here.